### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by nachtien**
Let's use snake case instead of camelcase for xml variables.

### CODE REVIEW

**Comment by nachtien** [comment]
@TravisJuul @casey-juul @JuulLabs/juul-eng-app

File: README.md [hunk 1]
```diff
@@ -60,13 +60,11 @@ View ID names (`android:id` in XML) should be all lowercase with underscores sep
...
-
-When using [Kotlin Android Extensions] to provide synthetic view properties, use the `as` keyword on [imports] to name
-properties according to [non-constant names Android Kotlin codestyle]. The name should be [camel case], postfixed with
-it's view type:
+The name should be [snake case].  Do your best to create a name that contains an underscore so it is distinguishable from local variables.  If a name cannot be created with an underscore, then postfix [_view] at the end.
 
 ```kotlin
-import kotlinx.android.synthetic.main.scan.dance_moves as danceMovesRecyclerView
+import kotlinx.android.synthetic.main.scan.dance_moves
+import kotlinx.android.synthetic.main.scan.login as login_view
```
Comment (ebabel) [inline README.md]: I was thinking we'd always use at least one `_` in the xml

**Review by ebabel — COMMENTED** [review]
I was thinking we'd always use at least one `_` in the xml. But w/e

**Comment by nachtien** [comment]
@ebabel Travis and I talked about the case where there might (hopefully never) be where a view wouldn't be snake cased, so we came up with a use case for that.  How do you feel about postfixing `_view` to the end of those properties

File: README.md [hunk 2]
```diff
@@ -53,20 +53,25 @@ val ipAddr = "127.0.0.1" // WRONG!
...
 When using [Kotlin Android Extensions] to provide synthetic view properties, use the `as` keyword on [imports] to name
-properties according to [non-constant names Android Kotlin codestyle]. The name should be [camel case], postfixed with
-it's view type:
+properties that do not have an underscore in their name, and append `_view` to the end
+
 
 ```kotlin
-import kotlinx.android.synthetic.main.scan.dance_moves as danceMovesRecyclerView
+import kotlinx.android.synthetic.main.scan.dance_moves
+import kotlinx.android.synthetic.main.scan.login as login_view
```
Comment (ghost) [inline README.md]: I think one of the benefits of going with the XML name is that we no longer have to go through the trouble of specifying `as` for the `import`.

So perhaps the requirement for having an `_` should be for the XML `id`?

File: README.md [hunk 3]
```diff
@@ -53,20 +53,25 @@ val ipAddr = "127.0.0.1" // WRONG!
...
+
+<Button
+    android:id="@+id/login"
+    />
 ```
 
 When using [Kotlin Android Extensions] to provide synthetic view properties, use the `as` keyword on [imports] to name
-properties according to [non-constant names Android Kotlin codestyle]. The name should be [camel case], postfixed with
-it's view type:
+properties that do not have an underscore in their name, and append `_view` to the end
```
Comment (casey-juul) [inline README.md]: @TravisJuul I thought you kinda agree with me on this one... or at least cringed  😛 

Either you should go with all views having "_<type>" or none. This feels like a hack :)

**Review by casey-juul — COMMENTED** [review]
Me being difficult on purpose aside, I really do feel like adding "_view" solely for the purpose of reinforcing the underbar decision is hacky. But this is also super low priority in the grand scheme of things, so whatever you guys decide is fine.

File: README.md [hunk 4]
```diff
@@ -53,20 +53,25 @@ val ipAddr = "127.0.0.1" // WRONG!
...
 <android.support.v7.widget.RecyclerView
     android:id="@+id/dance_moves"
     />
+
+<Button
+    android:id="@+id/login"
+    />
 ```
 
 When using [Kotlin Android Extensions] to provide synthetic view properties, use the `as` keyword on [imports] to name
```
Comment (ebabel) [inline README.md]: take out `as`

**Review by ebabel — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
I personally like using the XML ID in code (i.e. with underscores), but playing devil's advocate: lint complains about underscores:

screen shot 2018-05-31 at 1 43 18 pm

Although the lint only seems to catch the variables that came in from `findViewById`, but shows that we would be violating a standard in Android if we when with the "underscore in variable name" approach.

**Comment by twyatt** [comment]
Bump. I think we should merge this, as establishing this as a standard would be helpful, and we can hopefully later look into disabling the lint check.

> I really do feel like adding "_view" solely for the purpose of reinforcing the underbar decision is hacky

I agree with @casey-juul, although I think the rule of omitting `_view` postfix should be a recommendation (not a requirement). In most cases coming up with a 2 word description of a component should be pretty easy. In instances were it proves difficult, then as a last resort it should be fine to postfix the name with the component type (e.g. `_button` or `_view`).

If others agree, then maybe we should add a little more detail to the `README.md` changes in this PR?

**Comment by twyatt** [comment]
We agreed in-person that https://github.com/JuulLabs-OSS/kotlin-guides/pull/1#issuecomment-434167788 is a good approach. @nachtien, can you update the PR verbiage accordingly?

**Comment by ebabel** [comment]
I'm cool with either, slight preference for the '_hacky_' "adding "_view" solely for the purpose of reinforcing the underbar decision"

File: README.md [hunk 5]
```diff
@@ -53,20 +53,19 @@ val ipAddr = "127.0.0.1" // WRONG!
 
 ### View IDs
 
-View ID names (`android:id` in XML) should be all lowercase with underscores separating words.
+View ID names (`android:id` in XML) should be all lowercase with underscores separating words (snake_case).  It is required that every name will have at least one underscore that way it is distinguishable from local variables. In instances were it proves difficult, then as a last resort, it should be fine to postfix the name with the component type (e.g. _button or _view).
```
Comment (twyatt) [inline README.md]:
```suggestion
View ID names (`android:id` in XML) should be all lowercase with underscores separating words (snake_case). It is required that every name have at least one underscore to distinguish it from local variables. In instances were it proves difficult, then name may be postfixed with the component type (e.g. `_button` or `_view`).
```

**Review by twyatt — CHANGES_REQUESTED** [review]

**Review by twyatt — APPROVED** [review]

**Pull request merged**